cli/src/pcluster/config/cluster_config.py

```diff
@@ -2373,8 +2485,13 @@ def get_instance_types_data(self):
         result[instance_type_info.instance_type()] = instance_type_info.instance_type_data
         for queue in self.scheduling.queues:
             for compute_resource in queue.compute_resources:
-                instance_type_info = compute_resource.instance_type_info
-                result[instance_type_info.instance_type()] = instance_type_info.instance_type_data
+                if isinstance(compute_resource, SlurmComputeResource):
+                    instance_type_info = compute_resource.instance_type_info
+                    result[instance_type_info.instance_type()] = instance_type_info.instance_type_data
+                if isinstance(compute_resource, SlurmFlexibleComputeResource):
+                    for instance_type in compute_resource.instance_type_names:
```

Comment:
I'd add a property called:  `instance_type_list` in the base compute resource (like the one we have in the Queue), that returns a list with an item in the basic case or a list of instance types in the flexible case.

Reply:
I'm for calling the function "instance_type_list" and not "instance_type_names", to avoid introducing another naming. We're already using "instance_type_list" at queue level so we can have the same at compute resource level.